cmd/clusterctl/client/repository/metadata_client.go

```diff
@@ -92,7 +92,39 @@ func (f *metadataClient) Get(ctx context.Context) (*clusterctlv1.Metadata, error
 		return nil, errors.Wrapf(err, "error decoding %q for provider %q", metadataFile, f.provider.ManifestLabel())
 	}
 
-	//TODO: consider if to add metadata validation (TBD)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse current version %q for provider %q", f.version, f.provider.ManifestLabel())
+	}
+	if err := validateMetadata(obj, f.provider.ManifestLabel()); err != nil {
+		return nil, err
+	}
 
 	return obj, nil
 }
+
+// validateMetadata validates the metadata object structure.
+//
+// It checks if:
+// 1. The metadata has the correct apiVersion and kind.
+// 2. The metadata has at least one release series.
+//
+// Note: Version matching against releaseSeries is done later in `installer.go`.
+func validateMetadata(metadata *clusterctlv1.Metadata, providerLabel string) error {
+	// Check if metadata has the correct apiVersion and kind
+	if metadata.APIVersion != clusterctlv1.GroupVersion.String() {
+		return errors.Errorf("invalid provider metadata: unexpected apiVersion %q for provider %s (expected %q). Full metadata.yaml content is available in logs with --v=5",
```

Comment:
```suggestion
		return errors.Errorf("invalid provider metadata: unexpected apiVersion %q for provider %s (expected %q).",
```
This is not true anymore (same below)

Reply:
Thank you. I fixed.